src/mbgl/text/collision_tile.cpp

```diff
@@ -149,48 +167,71 @@ Box CollisionTile::getTreeBox(const Point<float>& anchor, const CollisionBox& bo
 }
 
 std::vector<IndexedSubfeature> CollisionTile::queryRenderedSymbols(const GeometryCoordinates& queryGeometry, float scale) {
-
     std::vector<IndexedSubfeature> result;
-    if (queryGeometry.empty()) return result;
+    if (queryGeometry.empty() || (tree.empty() && ignoredTree.empty())) {
+        return result;
+    }
 
+    // Account for the rounding done when updating symbol shader variables.
+    float zoom = util::log2(scale);
+    zoom = std::ceil(zoom * 10.0f) / 10.0f;
+    scale = std::pow(2.0f, zoom);
```

Comment:
I think this rounded scale needs to be used for the `visibleAtScale` but the unrounded scale needs to be used when scaling the box in `intersectsAtScale`. The rounded value is only used in the shaders to figure out when a label is visible. The actual size and position uses the more exact value.